### ISSUE TITLE
Implement Distillation (Phase 7)

### ISSUE THREAD

**Issue by noahgift** [paiml/entrenar#3]
# Phase 7: Knowledge Distillation

Implement knowledge distillation with temperature-scaled softmax and KL divergence loss.

## Tasks
- [ ] ENT-037: KD loss (temperature-scaled softmax) (16h)
- [ ] ENT-038: Multi-teacher ensemble distillation (24h)  
- [ ] ENT-039: Progressive distillation (layer-wise) (16h)
- [ ] ENT-040: Distillation effectiveness property tests (8h)

## Spec
```rust
pub struct DistillationLoss {
    temperature: f32,
    alpha: f32,  // Weight for distillation vs task loss
}

impl DistillationLoss {
    pub fn forward(
        &self,
        student_logits: &Tensor,
        teacher_logits: &Tensor,
        labels: &[usize],
    ) -> f32 {
        // Soft targets: KL(softmax(teacher/T) || softmax(student/T))
        // Hard targets: Cross-entropy(student, labels)
        // Combined: alpha * kl_loss + (1 - alpha) * ce_loss
    }
}
```

## Acceptance Criteria
- ✅ Temperature-scaled softmax KL divergence
- ✅ Multi-teacher ensemble distillation
- ✅ Layer-wise progressive distillation
- ✅ Property tests for distillation effectiveness
- ✅ All tests pass (>90% coverage)
- ✅ Clippy clean
- ✅ Comprehensive examples

Estimated: 64h total
